### PR TITLE
Use of RPS for better distributional regression

### PR DESCRIPTION
**Pull request by jonaslandsgesell**
## Motivation and Context

In this PR, we (Jonas Landsgesell and Pascal Knoll) implement the Ranked Probability Score (RPS) for fine-tuning TabPFN.

Ranked Probability Score (RPS) for distributional regression is a distance-sensitive scoring rule—unlike Cross Entropy (CE) loss, which is a local scoring rule. Therefore, RPS has advantages in the regression setting.

We would love to see how a TabPFN model pretrained with the RPS loss performs compared to a TabPFN model pretrained on the Cross Entropy loss. Preliminary evaluation on the regression finetuning example shows benefits of RPS over CE, when run with multiple seeds.

Below you can see our RPS vs. CE evaluation on a custom NN architecture (not TabPFN) on holdout test sets for different datasets:

Better MAE with RPS-trained NN than CE-trained NN:
image

Better R2 with RPS-trained NN than CE-trained NN:
image

Better RPS with RPS-trained NN than CE-trained NN:
image

General Literature:
* [Making and Evaluating Point Forecasts Tilmann Gneiting](https://arxiv.org/pdf/0912.0902)

* https://scoringrules.readthedocs.io/en/latest/theory.html

* [Rage against the mean](https://www.sciencedirect.com/science/article/abs/pii/S2452306221000824)

* [Gneiting, Strictly Proper Scoring Rules, Prediction, and Estimation](https://sites.stat.washington.edu/raftery/Research/PDF/Gneiting2007jasa.pdf )

## Public API Changes

-   [ x] No Public API changes
-   [ ] Yes, Public API changes (Details below)

---

## How Has This Been Tested?
Yes on the example finetuning script for regression (classification is unchanged)
---

## Checklist

-   [x ] The changes have been tested locally.
-   [ ] Documentation has been updated (if the public API or usage changes).
-   [ ] A entry has been added to `CHANGELOG.md` (if relevant for users).
-   [ ] The code follows the project's style guidelines.
-   [ ] I have considered the impact of these changes on the public API.

---

🎅🎄

### CODE REVIEW

**Comment by bejaeger** [comment]
Hi @jonaslandsgesell and Pascal!
Thanks again for the great work here. We took your suggestions and implemented it in a new PR #711 to be compatible with our refactored finetuning code. We also added the ranked logarithmic score as an option.

We’d love to get your eyes on this new version to make sure we’ve captured your intent correctly. Closing this PR now so we can centralize the discussion in #711 !
